File: src/main/java/com/ecommerce_backend/backend/core/domain/Order.java
package com.ecommerce_backend.backend.core.domain;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record Order(
        UUID id,
        Customer customer,
        List<OrderItem> items,
        LocalDateTime createdAt,
        OrderStatus status
) {
    public BigDecimal getTotal() {
        return items.stream()
                .map(item -> item.unitPrice().multiply(BigDecimal.valueOf(item.quantity())))
                .reduce(BigDecimal.ZERO, BigDecimal::add);
    }
}

File: ecommerce-frontend/src/domain/entities/Order.ts
import { Customer } from './Customer';

export interface Order {
  id: string;
  customer: Customer;
  items: OrderItem[];
  createdAt: string;
  status: OrderStatus;
}

export interface OrderItem {
  id: string;
  productId: string;
  quantity: number;
  unitPrice: number;
}

export type OrderStatus = 'PENDING' | 'PAID' | 'SHIPPED' | 'DELIVERED' | 'CANCELED';

export interface OrderRequest {
  customerId: string;
  items: OrderItemRequest[];
}

export interface OrderItemRequest {
  productId: string;
  quantity: number;
  unitPrice: number;
}

export interface OrderStatusUpdateRequest {
  status: OrderStatus;
}